enhancements/helm3/console.md

```diff
@@ -0,0 +1,144 @@
+
+## Motivation
+
+Helm is a Kubernetes package manager.  Helm 3.0 is a major release of helm which brings in a rich set of features and addresses major security concerns around tiller.  
+
+Red hat Openshift wants to bring Helm based content support to Openshift 4.4 Developer Catalog along with Operators to strengthen the helm based ecosystem.
+
+![Developer Catalog](../helm3/assets/dev-catalog.png)
+
+### Goals
+
+* Provide RESTful API for managing Helm charts and releases
+* Support disconnected installs
+* Support easy management of available charts, aggregation from multiple sources and filtering
+  
+### Non-Goals
+
+* Infrastructure for serving the default chart repository
+* Process for curating charts within the default chart repository
+  
+## Proposal
+  
+### Charts in the Developer Catalog
+
+The charts that would show up in the Developer Catalog will be powered by a [standard](https://helm.sh/docs/topics/chart_repository) helm charts' repository service.
+
+The default chart repository will be served from [Red Hat public GitHub repository](https://github.com/redhat-developer/redhat-helm-charts).
+
+New charts will be added or existing curated by submitting PRs against the above GitHub repo.
+
+### How would the UI discover the charts
+
+1. The UI would invoke `/api/helm/charts/index.yaml` endpoint to get [the list of all available charts](https://helm.sh/docs/topics/chart_repository/#the-index-file) so that the available charts can be rendered in the developer catalog. 
+
+2. The above endpoint would proxy requests to the configured charts repository
+
+In the initial phase, the charts repository would be a served of [redhat-helm-charts](https://redhat-developer.github.io/redhat-helm-charts) public GitHub repo.
+
+![Helm Charts Repo Service](../helm3/assets/charts-repo.png)
+
+### How would disconnected installs work 
+
+1. The user would need to 'clone' the content of the chart repository over the fence
+
+   * The public GitHub repository could be cloned into inside-the-network GitHub or Gitlab instance and configured to serve static content ( "Pages" ).
+   * The content of the chart repository could be crawled and served using a (containerized or external) HTTP server, e.g. nginx
+
+2. The URL serving the above static content would need to be passed to chart repository proxy running inside the cluster. 
+
+### Configuring Helm Chart Repository location
+
+Configuring Helm repository location could be modeled similar to [`OperatorSource`](https://github.com/operator-framework/operator-marketplace/blob/7d230952a1045624b7601b4d6e1d45b3def4cf76/deploy/crds/operators_v1_operatorsource_crd.yaml). 
+Due to future federated usecases ([ODC-2994](https://issues.redhat.com/browse/ODC-2994))), a cluster admin should be able to declare multiple chart repositories.
+
+A cluster-scoped, top-level `HelmRepository` resource)[https://github.com/openshift/api/tree/master/config/v1] should be added:
+
+```yaml
+apiVersion: config.openshift.io/v1
```

Comment:
I would not expect it in this group.  If this is only for consumption by the console, I would expect it localized in some way.  Does any other team need to interact with this?

Reply:
What do you suggest having instead as the group? 

At the moment, it is only the Console operator controller which consumes the information to configure the console Deployment.

Conceptually, it is not part of Console. In future, there might be other consumers too.

Reply:
We are aiming at cluster wide configuration, as we agreed with @derekwaynecarr the last week.

Reply:
> We are aiming at cluster wide configuration, as we agreed with @derekwaynecarr the last week.

A cluster-wide configuration that is later accessed by a proxy in in the console reads really unusually.

Unless the proxy is just a single way of consuming this, it sounds like you're creating a console API as opposed to the peer of an OperatorHub source.

Similarly, since only one instance is allowed, I would expect multiple possible sources for charts.  This API appears to be built to have multiple instances instead of a single cluster instance.

Reply:
@pedjak I know we met prior, and I am now caught up on the enhancement.

instead of config.openshift.io, should we instead just a dedicated helm.openshift.io group?

if its in config.openshift.io, i agree  that w/ @deads2k that it needs to look more like OperatorHub (singular).

do you foresee other helm related APIs getting added in the future that would warrant a helm.openshift.io group?

Reply:
> instead of config.openshift.io, should we instead just a dedicated helm.openshift.io group?

@derekwaynecarr Is the implicit restriction of `config.openshit.io` group that we can have single CR instance? I am not against moving it to `helm.openshift.io`, but then we are going to loose the ability to edit/configure it via UI, i.e. Administration/Cluster Settings/Global Configuration, correct?

>  do you foresee other helm related APIs getting added in the future that would warrant a helm.openshift.io group?

Not really - we have already added IMHO the majority of needed API. However, if we would like to keep a few placeholders for future config options, and if `config.openshift.io` can handle only singletons, we could define then `Helm` top-level CR, that would contain for the moment a list of chart repos only, something like this:

```yaml
apiVersion: config.openshift.io/v1
kind: Helm
metadata:
  name: cluster
spec:
  chartRepositories:
    - url: http://my.chart-repo.org/stable

      # optional and only needed for UI purposes
      displayName: myChartRepo

      # optional and only needed for UI purposes
      description: my private chart repo

      # set to true if need to be disabled temporarly
      disabled: true
```

Does it make sense @deads2k ?

Reply:
> @derekwaynecarr Is the implicit restriction of `config.openshit.io` group that we can have single CR instance?

Yes.  excepting clusteroperators and clusterversion, everything else is a single instance named cluster.



> I am not against moving it to `helm.openshift.io`, but then we are going to loose the ability to edit/configure it via UI, i.e. Administration/Cluster Settings/Global Configuration, correct?

Seems like you'd just update the glass to edit them in the same spot.